gtsam/sfm/DsfTrackGenerator.h

```diff
@@ -0,0 +1,188 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file DsfTrackGenerator.h
+ * @date July 2022
+ * @author John Lambert
+ * @brief Identifies connected components in the keypoint matches graph.
+ */
+
+#pragma once
+#include <Eigen/Core>
+
+#include <gtsam/base/DSFMap.h>
+#include <gtsam/geometry/Point2.h>
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <optional>
+#include <vector>
+
+namespace gtsam {
+
+typedef DSFMap<IndexPair> DSFMapIndexPair;
+typedef std::pair<size_t, size_t> ImagePair;
+typedef Eigen::MatrixX2i CorrespondenceIndices; // N x 2 array
+
+//struct Keypoints;
+using KeypointCoordinates = Eigen::MatrixX2d;
+
+
+struct Keypoints {
+  KeypointCoordinates coordinates;
+  // typedef'd for Eigen::VectorXd
+  boost::optional<gtsam::Vector> scales;
+  boost::optional<gtsam::Vector> responses;
+
+  Keypoints(const gtsam::KeypointCoordinates& coordinates): coordinates(coordinates) {}; // boost::none
+};
+
+using KeypointsList = std::vector<Keypoints>;
+using KeypointsVector = std::vector<Keypoints>; // TODO(johnwlambert): prefer KeypointsSet?
+using MatchIndicesMap = std::map<ImagePair, CorrespondenceIndices>;
+
+
+// @param camera index
+// @param 2d measurement
+// Implemented as named tuple, instead of std::pair (like SfmMeasurement in SfmTrack.h)
+struct NamedSfmMeasurement {
+  size_t i;
+  gtsam::Point2 uv;
+
+  NamedSfmMeasurement(size_t i, gtsam::Point2 uv) : i(i), uv(uv) {}
```

Comment:
Don’t we need it if we want to expose this type on the Python side?

Reply:
Im not sure, I wonder if pybind adds a default constructor? 
Otherwise you should be able to do 
```
measurement = NamedSfMMeasurement()
measurement.i = 0
measurement.uv = (0, 1)
```

Reply:
From my recollection the constructor would have to be explicit in the .i file to be registered. I think the keyword arg constructor is a bit cleaner (1 line vs 3 lines) : - )